### PR TITLE
Simplestring

### PR DESCRIPTION
**Pull request by arstrube**
Additional tests and a small fix for SimpleString

### CODE REVIEW

File: tests/SimpleStringTest.cpp [hunk 1]
```diff
@@ -257,6 +257,12 @@ TEST(SimpleString, countTogether)
     LONGS_EQUAL(4, str.count("ha"));
 }
 
+TEST(SimpleString, countEmptyString)
+{
+    SimpleString str("hahahaha");
+    LONGS_EQUAL(1, str.count(""));
```
Comment (basvodde) [inline tests/SimpleStringTest.cpp]: Uh, why?

Comment (arstrube) [inline tests/SimpleStringTest.cpp]: What would you rather expect? How many times is the empty string contained in a string? :P

Behavior so far was some very large unsigned long integer value, possibly MAX_ULONG. This lead to problems with the empty string in replace().

Comment (arstrube) [inline tests/SimpleStringTest.cpp]: This lead to a noticable delay (30 seconds or so), as well as the replace function not working properly for empty strings (the else part was never executed).

Comment (arstrube) [inline tests/SimpleStringTest.cpp]: Returning zero would be fine by me and work okay. But I could't give you any better justification for returning zero than for returning 1...

After all, how many times is the empty string contained in a string ;-).

Comment (basvodde) [inline tests/SimpleStringTest.cpp]: I would have thought 0 :) What does strstr do?

Comment (arstrube) [inline tests/SimpleStringTest.cpp]: strstr returns the entire first string, if the string to search for is the empty string, apparently. Knowning that doesn't really help, does it... I'm fine with zero.

Comment (arstrube) [inline tests/SimpleStringTest.cpp]: This behavior of strstr spells trouble for the while loop in count, I guess.

File: src/CppUTest/SimpleString.cpp [hunk 2]
```diff
@@ -194,7 +193,7 @@ size_t SimpleString::count(const SimpleString& substr) const
 {
     size_t num = 0;
     char* str = buffer_;
-    while ((str = StrStr(str, substr.buffer_))) {
+    while (*str && (str = StrStr(str, substr.buffer_))) {
```
Comment (basvodde) [inline src/CppUTest/SimpleString.cpp]: strstr returns the same string as it passed. That would suggest that the count would be equal to size(). What do you think about that? Which means the *str shouldn't be needed....

Comment (basvodde) [inline src/CppUTest/SimpleString.cpp]: Nevermind above comment, str isn't the substr but the buffer_ It is early morning (5:10am)

Comment (arstrube) [inline src/CppUTest/SimpleString.cpp]: :-)

**Pull request merged**